apple/Sources/FerrostarMapLibreUI/Views/Overylays/PortraitNavigationOverlayView.swift

```diff
@@ -45,44 +47,58 @@ struct PortraitNavigationOverlayView: View, CustomizableNavigatingInnerGridView
     }
 
     var body: some View {
-        VStack {
-            if case let .navigating(_, _, _, _, progress: progress, _, visualInstruction: visualInstruction,
-                                    _) = navigationState?.tripState,
+        ZStack(alignment: .top) {
```

Comment:
The layout changes here are non-trivial.

We want to be able to expand the instruction view *over top* of the other content when it's expanded. But when it's not expanded, we want it to behave as if it's in a vstack with the NavigationInnerGridView.

My solution was to put the InstructionView in a ZStack above the grid, and then plumb out the InstructionView's "unexpanded height" to push down the NavigatingInnerGridView.

I struggled a bit with implementing this layout in SwiftUI. It's something I think would be natural to express in old school iOS layout constraints. Maybe there's a better way?

Reply:
Yeah, I'm not aware of a better way offhand either. It's one of the areas where SwiftUI and others struggle compared to springs-and-struts or auto layout.

Reply:
There are a few conflicts unfortunately with the main branch since @Archdoog just cleaned up a bunch of stuff to clean up the pattern matching mess 😅